src/libcadet/model/ModelSystemImpl.cpp

```diff
@@ -280,6 +280,26 @@ unsigned int ModelSystem::numPureDofs() const CADET_NOEXCEPT
 	return dofs;
 }
 
+std::vector<int> ModelSystem::getModelSlice() const CADET_NOEXCEPT
```

Comment:
Should this not take the unit index as input argument and then only return the slice of the model for that particular index? Or does it return a list/table of slices for all units?

Reply:
@schmoelder actually in this function I am using class "IUnitOperation", which contains the information about the number of models or unit operations. I just iterate over that information like this:
`for (IUnitOperation* m : _models)`
and calculate the starting index and stride of last state vector related to each unit operation. 
This function returns a integer vector containing information as follows:
Index and stride related to any UNIT_OPERATION i is stored at [2*i,2*i+1] location of the vector, where i=0,...N-1 (N is the number of unit operations).

Reply:
@sleweke @schmoelder @Immudzen (Suggestion post), while implementing the last_state_y for each unit operation I got this question in mind why can't use the current scope of `_writeLastState` to output the Last_State of each unit operaton alongwith global Last_State? 
So, with one output selection we can get the global LAST_STATE_Y and LAST_STATE_Y_UNIT_OPERATION_00i or is it not feasible?
Following you can find snippet of the original code for writing LAST_STATE_Y in the output file of CADET, so all the addition in the code related to this issue can also go in this section. 
```
if (_writeLastState)
		{
			unsigned int len = 0;
			double const* const lastY = _sim->getLastSolution(len);
			double const* const lastYdot = _sim->getLastSolutionDerivative(len);
			writer.vector("LAST_STATE_Y", len, lastY);
			writer.vector("LAST_STATE_YDOT", len, lastYdot);
                 }
```

Reply:
Regarding the function, `getModelSlice()`:
1. The function does not return the actual slice but an offset in the global state vector. Thus, i'd suggest naming it `modelOffsetInGlobalState()` or something similar.
2. The information required for this function is already stored in the member variable `_dofOffset`. However, the last entry of this vector additionally contains the offset to the connection DOFs. Thus, it may not be wise to simply return this vector.  Instead, one could return a pointer to this offset array (i.e., `_dofOffset.data()`) such that the callee has to assume that the array is of length `nUnits`.